File: backend/__SLUG__/src/domains/health/datasource.go
package health

// Datasource handles data persistence for health domain
// In this case, health is stateless, but this demonstrates the pattern
type Datasource struct {
	// In a real scenario, this might have DB connection, cache, etc.
}

// NewDatasource creates a new datasource instance
func NewDatasource() *Datasource {
	return &Datasource{}
}

// GetSystemInfo retrieves system information (placeholder for demonstration)
func (ds *Datasource) GetSystemInfo() (string, string, error) {
	// In a real app, this might query a database or system metrics
	return "go-backend", "1.0.0", nil
}

File: backend/__SLUG___node/src/domains/health/datasource.ts
/**
 * Datasource for the health domain. Health is stateless, but this class
 * demonstrates the datasource pattern used across every domain.
 */
export class Datasource {
  /** Returns system info (placeholder for demonstration). */
  getSystemInfo(): { service: string; version: string } {
    return { service: "node-backend", version: "1.0.0" };
  }
}
